### PR TITLE
Vuepress 2.x version

### LINKED ISSUE #1239: Search bar not working sometimes (with new added FAQs)
Looking up `WHAT DOES THE PRIVACY PROGRESS MEAN?` in the search bar does not return anything.

### PR DESCRIPTION
**Pull request by CAnorbo**
Updated Vuepress 2.x version.
Fixes #1239  #1238  #1211 #1339 #1342

@yahiheb @MarnixCroes @molnard

### CODE REVIEW

**Review by yahiheb — CHANGES_REQUESTED** [review]
Your branch is not up to date with master.

**Comment by CAnorbo** [comment]
> Your branch is not up to date with master.

Updated could you check the changes?

File: docs/.vuepress/enhanceApp.js [hunk 1]
```diff
@@ -59,4 +59,12 @@ export default ({ router }) => {
       })
     })
   }
-}
+  router.addRoutes([
+    { path: '/building-wasabi/HardwareWalletTestingGuide.html', redirect: '/' },
+    { path: '/building-wasabi/FalsePositive.html', redirect: '/' },
+    { path: '/why-wasabi/LegalConcern.html#:~:text=be%20the%20case.-,Overview,funds%20with%20an%20obfuscated%20history.', redirect: '/' },
+    { path: '/building-wasabi/ToDo.html', redirect: '/' },
+    { path: '/building-wasabi/Dojo.html', redirect: '/' },
```
Comment (yahiheb) [inline docs/.vuepress/enhanceApp.js:67]: The Dojo file was removed long ago.

File: docs/.vuepress/variables.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 // Add custom variables here:
 const variables = {
-  currentVersion: '2.0.1.4',
-  dotnetVersion: '6.0',
+  currentVersion: '1.1.13',
+  dotnetVersion: '5.0',
```
Comment (yahiheb) [inline docs/.vuepress/variables.js:4]: This is not up to date.

File: package.json [hunk 3]
```diff
@@ -1,43 +1,23 @@
...
+    "docs:dev": "vuepress dev docs",
+    "docs:build": "vuepress build docs"
   },
-  "license": "MIT",
-  "bugs": {
-    "url": "https://github.com/zkSNACKs/WasabiDoc/issues"
-  },
-  "homepage": "https://github.com/zkSNACKs/WasabiDoc#readme",
+  "author": "",
+  "license": "ISC",
```
Comment (yahiheb) [inline package.json:12]: What is this?

**Review by yahiheb — CHANGES_REQUESTED** [review]
`Logo_with_text.png` is not used anywhere.

Did you mean to make all the changes in `package-lock.json` and in `package.json`?

I do not understand what this PR does, nor how to review it, and from the changes in `variables.js` I believe the branch is still not up-to-date.

I guess it is better to fork the repository again, and then checkout a new branch and make your changes, then make a new PR.

**Comment by molnard** [comment]
Concept ACK, CI is failing, can you fix it?

**Comment by CAnorbo** [comment]
> Concept ACK, CI is failing, can you fix it?

Fixed CL

**Comment by BTCparadigm** [comment]
images (17)

**Comment by molnard** [comment]
I suggest to prioritize this PR before anything else in the repo, otherwise, you will get conflicts again and again.

**Review by yahiheb — CHANGES_REQUESTED** [review]
@CAnorbo  What is the purpose of this PR? Why do we need to upgrade to Vuepress 2?

I get this error when executing `npm start`:

> TypeError: Cannot destructure property 'themeConfig' of 'ctx' as it is undefined.
>     at module.exports (C:\Users\chihe\OneDrive\Bureau\WasabiDoc\node_modules\@vuepress\theme-default\index.js:7:11)
>     at file:///C:/Users/chihe/OneDrive/Bureau/WasabiDoc/docs/.vuepress/config.bundled_1665960827483.mjs:19:10
>     at ModuleJob.run (node:internal/modules/esm/module_job:193:25)
>     at async Promise.all (index 0)
>     at async ESMLoader.import (node:internal/modules/esm/loader:533:24)
>     at async importModuleDynamicallyWrapper (node:internal/vm/module:438:15)
>     at async extractResult (C:\Users\chihe\OneDrive\Bureau\WasabiDoc\node_modules\bundle-require\dist\index.js:263:13)
>     at async loadConfig (C:\Users\chihe\OneDrive\Bureau\WasabiDoc\node_modules\@vuepress\core\lib\node\loadConfig.js:30:21)
>     at async App.resolveConfigAndInitialize (C:\Users\chihe\OneDrive\Bureau\WasabiDoc\node_modules\@vuepress\core\lib\node\App.js:64:24)
>     at async App.process (C:\Users\chihe\OneDrive\Bureau\WasabiDoc\node_modules\@vuepress\core\lib\node\App.js:103:5)
>     at async dev (C:\Users\chihe\OneDrive\Bureau\WasabiDoc\node_modules\@vuepress\core\lib\index.js:17:3)

**Comment by CAnorbo** [comment]
> @CAnorbo What is the purpose of this PR? Why do we need to upgrade to Vuepress 2?
> 
> I get this error when executing `npm start`:
> 
> > TypeError: Cannot destructure property 'themeConfig' of 'ctx' as it is undefined.
> > at module.exports (C:\Users\chihe\OneDrive\Bureau\WasabiDoc\node_modules@vuepress\theme-default\index.js:7:11)
> > at file:///C:/Users/chihe/OneDrive/Bureau/WasabiDoc/docs/.vuepress/config.bundled_1665960827483.mjs:19:10
> > at ModuleJob.run (node:internal/modules/esm/module_job:193:25)
> > at async Promise.all (index 0)
> > at async ESMLoader.import (node:internal/modules/esm/loader:533:24)
> > at async importModuleDynamicallyWrapper (node:internal/vm/module:438:15)
> > at async extractResult (C:\Users\chihe\OneDrive\Bureau\WasabiDoc\node_modules\bundle-require\dist\index.js:263:13)
> > at async loadConfig (C:\Users\chihe\OneDrive\Bureau\WasabiDoc\node_modules@vuepress\core\lib\node\loadConfig.js:30:21)
> > at async App.resolveConfigAndInitialize (C:\Users\chihe\OneDrive\Bureau\WasabiDoc\node_modules@vuepress\core\lib\node\App.js:64:24)
> > at async App.process (C:\Users\chihe\OneDrive\Bureau\WasabiDoc\node_modules@vuepress\core\lib\node\App.js:103:5)
> > at async dev (C:\Users\chihe\OneDrive\Bureau\WasabiDoc\node_modules@vuepress\core\lib\index.js:17:3)

It's fixing 3 issues that I couldn't do on the 1.x version.
Check your node version and make sure it's higher than 16.

**Comment by yahiheb** [comment]
> Check your node version and make sure it's higher than 16.

I updated Node.js to the version 18.12.0 and I still get the same error and I can't run this locally.

**Comment by CAnorbo** [comment]
> > Check your node version and make sure it's higher than 16.
> 
> I updated Node.js to the version 18.12.0 and I still get the same error and I can't run this locally.

This should work:
npm install
npm update
npm run start

**Comment by yahiheb** [comment]
> This should work:
> npm install
> npm update
> npm run start

Yes it worked.

**Review by yahiheb — CHANGES_REQUESTED** [review]
@CAnorbo Sorry for letting this PR trail for long, it is much better than what I have initially thought, very good job.

Can you fix the conflicts?

**Comment by yahiheb** [comment]
I have noticed that you remove `:::details` so the FAQs are not longer expandable. Why is that?

**Comment by yahiheb** [comment]
@CAnorbo I deleted `.idea` folder to leave it out of source control.

**Comment by yahiheb** [comment]
This PR also fixes #1342 and #1339 because expandable items were removed https://github.com/zkSNACKs/WasabiDoc/pull/1247#issuecomment-1320968300

**Review by yahiheb — CHANGES_REQUESTED** [review]
**Must be fixed:**

- Variables like current version and zkSNACKs pubkey are not displayed correctly:

  doc1

**Nice to have:**

- Read the embedded videos on the docs and don't redirect to youtube.
- Expand/collapse FAQs items. https://github.com/zkSNACKs/WasabiDoc/pull/1247#issuecomment-1320968300
- Change the ugly looking default image of the embedded videos:

  doc0

**Comment by jesseaam** [comment]
On the index page, when I change from dark to light, some of the text is no longer visible.

<details>
<summary>Expand screenshots</summary>

## Dark mode view
image

<br><br>
`open-source` `non-custodial` and `privacy-focused`  you can't see now in light-mode.
## Light mode view
image
</details>

**Comment by jesseaam** [comment]
"Edit this page" link is broken.

Screen Shot 2022-11-21 at 10 47 06 AM

### leads me to here
image

**Comment by jesseaam** [comment]
Also, can we remove the Contributors section?

**Comment by CAnorbo** [comment]
> **Must be fixed:**
> 
> * Variables like current version and zkSNACKs pubkey are not displayed correctly:
>   doc1 **=> Tried to resolve it but no solution yet.**
> 
> **Nice to have:**
> 
> * Read the embedded videos on the docs and don't redirect to youtube. **=> This won't work. Spent a lot of hours on this topic before.**
> * Expand/collapse FAQs items. [#1247 (comment)](https://github.com/zkSNACKs/WasabiDoc/pull/1247#issuecomment-1320968300) **=> They were removed by me because of the better readability.**
> * Change the ugly looking default image of the embedded videos:
>   doc0 **=> This problem comes from youtube, we have to set a good image on youtube for this video. It's not a code problem.**

**Comment by CAnorbo** [comment]
> > > On the index page, when I change from dark to light, some of the text is no longer visible.
> > > Expand screenshots
> > 
> > 
> > fixed.
> 
> @CAnorbo It is not fixed for me.

This should fix => https://github.com/zkSNACKs/WasabiDoc/pull/1247/commits/a50a690ee0fae729197d6241c91397c03e1470e8

**Comment by yahiheb** [comment]
> This should fix => [a50a690](https://github.com/zkSNACKs/WasabiDoc/commit/a50a690ee0fae729197d6241c91397c03e1470e8)

That is what I have tested, and it is not fixed.

**Review by jesseaam — APPROVED** [review]
Tested ACK

**Review by yahiheb — CHANGES_REQUESTED** [review]
I still have the same issue as @jesseaam here https://github.com/zkSNACKs/WasabiDoc/pull/1247#issuecomment-1322241717
I cannot see some text in the light mode:

sCapture

**Comment by yahiheb** [comment]
> I still have the same issue as @jesseaam here https://github.com/zkSNACKs/WasabiDoc/pull/1247#issuecomment-1322241717
I cannot see some text in the light mode:

This is fixed.

**Comment by jesseaam** [comment]
The videos now open up a new tab in YouTube. They used to be embedded and play within the docs. Is this the new desired behavior?

**Comment by jesseaam** [comment]
Otherwise, everything else looks good on my end. Nice job everybody!

**Comment by yahiheb** [comment]
> I can have a look but I think it may make sense to ask in [Discussions](https://github.com/vuepress/vuepress-next/discussions) first to save some time. If we don't have a response in a few days, we can try ourselves.

@CAnorbo Do you mind asking about that as @kiminuo suggested?

**Comment by CAnorbo** [comment]
> > I can have a look but I think it may make sense to ask in [Discussions](https://github.com/vuepress/vuepress-next/discussions) first to save some time. If we don't have a response in a few days, we can try ourselves.
> 
> @CAnorbo Do you mind asking about that as @kiminuo suggested?

@yahiheb could you try this plugin? https://vuepress-theme-hope.github.io/v2/components/ on the Discussion the community recommended this one, but I can't run it for some reason.

File: README.md [hunk 4]
```diff
@@ -86,22 +86,24 @@ They can be embedded via the following markdown tags:
...
 And the index of the video of a playlist (e.g. starts the third video):
 
 ```md
-@[youtubePlaylist](PLAYLIST_ID,VIDEO_ID)
+[![Title of the playlist](https://img.youtube.com/vi/VIDEO_ID/maxresdefault.jpg)](https://youtu.be/VIDEO_ID?list=PLAYLIST_ID)
 ```
 
+If the image doesn't appear, that usually happens with older videos, use hqdefault.jpg instead of maxresdefault.jpg
+
+
```
Comment (yahiheb) [inline README.md:106]:
```suggestion
```

**Review by yahiheb — CHANGES_REQUESTED** [review]

**Comment by yahiheb** [comment]
> > > I can have a look but I think it may make sense to ask in [Discussions](https://github.com/vuepress/vuepress-next/discussions) first to save some time. If we don't have a response in a few days, we can try ourselves.
> > 
> > 
> > @CAnorbo Do you mind asking about that as @kiminuo suggested?
> 
> @yahiheb could you try this plugin? https://vuepress-theme-hope.github.io/v2/components/ on the Discussion the community recommended this one, but I can't run it for some reason.

That is not a blocker, we can fix that in a follow up PR.

**Review by yahiheb — APPROVED** [review]
tACK. Good job guys

**Pull request merged**